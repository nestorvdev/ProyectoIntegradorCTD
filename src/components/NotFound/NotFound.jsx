import React from "react";
import Styles from "./Styles.module.css";
import { Link } from "react-router-dom";

function NotFound() {
    return (
        <section className={Styles.delimiter}>
            <div className={Styles.box}>
                <div className={Styles.titleBox}>
                    <p className={Styles.title}>Error 404</p>
                    <h2>No se ha encontrado la página solicitada</h2>
                </div>
                <Link to={`/`}>
                    <button className={Styles.button}>Página Principal</button>
                </Link>
            </div>
        </section>
    )
}

export default NotFound;
