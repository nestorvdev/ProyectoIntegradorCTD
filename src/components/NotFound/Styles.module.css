.delimiter{
    align-items: center;
    display: flex;
    flex-direction: row;
    justify-content: center;
    width: 100%;
    height: 300px;
    margin-top: 5%;
    
}

.box {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;   
    height: 70%;
}

.titleBox {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;  
    height: 40%;
    height: 40%;
}

.delimiter .box p.title {
    font-size: var(--heading1);
    font-weight: bold;
    margin-bottom: 1pt;
    height: 100%;
}

.button {
    background-color: white;
    border-radius: 5px;
    border: 2px solid  var(--primary-color);
    background-color: var(--primary-color);
    color: white;
    font-size: var(--button1);
    font-weight: var(--font-bold);
    padding: 5pt;
    width: clamp(164px,22.2vw,206px);
    transition: 0.5s;
}
  

.button:hover {
    transform: scale(1.1);
    cursor: pointer;
}

.button:active {
    transform: scale(0.85); 
}