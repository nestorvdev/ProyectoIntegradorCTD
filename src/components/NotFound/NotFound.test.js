import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import NotFound from "./NotFound";

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe("Probando el componente <NotFound/>", () => {
    let wrapper;
    
    beforeEach(() => {
        wrapper = shallow(<NotFound/>)
    });

    it("Deberia mostrar <NotFound/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

})