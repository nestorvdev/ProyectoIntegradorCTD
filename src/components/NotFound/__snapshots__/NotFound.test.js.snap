// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <NotFound/> Deberia mostrar <NotFound/> correctamente 1`] = `
<section
  className="delimiter"
>
  <div
    className="box"
  >
    <div
      className="titleBox"
    >
      <p
        className="title"
      >
        Error 404
      </p>
      <h2>
        No se ha encontrado la página solicitada
      </h2>
    </div>
    <Link
      to="/"
    >
      <button
        className="button"
      >
        Página Principal
      </button>
    </Link>
  </div>
</section>
`;
